src/ToonFormat/Internal/Decode/PathExpansion.cs

```diff
@@ -0,0 +1,193 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using ToonFormat.Internal.Shared;
+
+namespace ToonFormat.Internal.Decode
+{
+    /// <summary>
+    /// Path expansion logic for dotted keys per SPEC §13.4
+    /// </summary>
+    internal static class PathExpansion
+    {
+        private static readonly Regex IdentifierPattern = new Regex(@"^[A-Za-z_][A-Za-z0-9_]*$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Expands dotted keys in a JsonObject into nested structures.
+        /// Example: {"a.b.c": 1} -> {"a": {"b": {"c": 1}}}
+        /// </summary>
+        /// <param name="obj">The object to expand</param>
+        /// <param name="strict">Whether to throw on conflicts</param>
+        /// <param name="quotedKeys">Set of keys that were originally quoted (should not be expanded)</param>
+        public static JsonObject ExpandPaths(JsonObject obj, bool strict, HashSet<string>? quotedKeys = null)
+        {
+            var result = new JsonObject();
+
+            foreach (var kvp in obj)
+            {
+                var key = kvp.Key;
+                var value = kvp.Value;
+
+                // Skip expansion for quoted keys (they should remain as literal dotted keys)
+                bool wasQuoted = quotedKeys != null && quotedKeys.Contains(key);
+
+                // Check if key contains dots and is eligible for expansion
+                if (!wasQuoted && key.Contains('.') && IsExpandable(key))
```

Comment:
Use constants where possible:

```suggestion
                if (!wasQuoted && key.Contains(Constants.DOT) && IsExpandable(key))
```

Reply:
Fixed in commit 1fe8fde8